azext_edge/tests/utility/test_az_client_unit.py

```diff
@@ -0,0 +1,40 @@
+# coding=utf-8
+# ----------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License file in the project root for license information.
+# ----------------------------------------------------------------------------------------------
+
+import pytest
+from ..generators import generate_random_string
+
+AZ_CLIENT_PATH = "azext_edge.edge.util.az_client"
+
+
+@pytest.mark.parametrize("done", [True, False])
+def test_wait_for_terminal_state(mocker, done):
+    # could be fixture with param
+    sleep_patch = mocker.patch(f"{AZ_CLIENT_PATH}.sleep")
```

Comment:
you did create a sleep mock... why not use it here? (I also do have some mocks that take in an optional path arg)

Reply:
That sleep mock is for a different test module (orchestration/base) and used multiple times. In this module there is once test that uses it.